### DISCUSSION TITLE
Does Magda support data access upon request?

### DISCUSSION THREAD

**Discussion by Duragtan** [magda-io/magda#3470] [Q&A] [answered]
Hello all,

I am wondering if Magda might be the right tool for my purposes. Can someone tell me if the following scenario is supported by Madga:
>    Person A has authored a data set, which he/she cannot easily disclose. Person A uploads the data set to Magda, but as a private data set that others can only see the meta-data about.
>    Now Person B is interested in working with the data. He/she contacts Person A via a web-form in Magda, requesting data access. Person B can now decide whether or not Person B should get access to the data set. If approved, Person B can access the full data on Magda for a defined period. Magda logs the entire interaction between the two Persons.
>    Occasionally, Person A might only want to put an entry into Magda describing the data set, but without actually uploading the data to Magda (a pure meta-data entry). Would the process described above still work for a pure meta-data entry. (In that case, the two would, of course, have to find some other means for actually transferring the data.)

Maybe there is an extension to Magda allowing for this, if not alreay possible with Madga by itself?

Best

Soraltan

**Comment by t83714** [comment, marked as answer]
Hi @Duragtan,

I think it's doable with Magda's built-in dataset data model.
Magda's dataset data model comes with two parts:
- distributions: a collection of records that represent the data files/resources/ APIs that the dataset contains.
- dataset: the main metadata record describe the dataset itself and also serve as the container of all resource / data files (via distributions) of the dataset.

The dataset / distribution records created by the built-in metadata authoring UI tool will set the permission of the dataset record and associated distribution records to the same permission.

You can (through the APIs) alternate the permission of distribution records (of a dataset) to achieve that:
- anonymous users (or nominated roles) can view the dataset record (dataset description)
- but only selected users can have access to the actual data file via the distribution records.

I guess you would want to maintain some sort of dataset approval transaction records as well. 
You can actually define custom data aspects to shape your own transaction records via APIs (see [aspect-related APIs](https://dev.magda.io/api/v0/apidocs/index.html#api-Registry_Aspects-PostV0RegistryAspects)).

To understand Magda's data model more, you might be interested in this tutorial repo:

https://github.com/magda-io/magda-brown-bag#tasks
